wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -285,10 +285,11 @@ def _etag_from_obj(
         ]
     ) -> ETag:
         etag: ETag
-        if hasattr(obj, "e_tag"):
-            etag = obj.e_tag[1:-1]  # escape leading and trailing quote
+        if hasattr(obj, "meta") and obj.meta.data is None:
+            # obj.meta.data is None for ObjectVersion while using s3.ObjectVersion()
+            etag = obj.get()["ETag"][1:-1]  # escape leading and trailing
```

Comment:
hm i just saw this, I don't think you want to call "get()" as it will download the object https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/objectversion/get.html

Reply:
You should use `head()` instead of `get()` here to fetch only the metadata https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/objectversion/head.html